IdentityCore/tests/MSIDDeviceInfoTests.m

```diff
@@ -246,7 +247,7 @@ - (void)testJsonDictionary_whenDeserialize_shouldGenerateCorrectJson {
         MSID_BROKER_WPJ_STATUS_KEY : @"joined",
         MSID_BROKER_BROKER_VERSION_KEY : @"1.2.3",
         MSID_ADDITIONAL_EXTENSION_DATA_KEY: @"{\"dict\":{\"key\":\"value\"},\"feature_flag1\":1,\"token\":\"\"}",
-        MSID_EXTRA_DEVICE_INFO_KEY:@"{\"mdm_id\":\"mdmId\",\"object_id\":\"objectId\"}"
+        MSID_EXTRA_DEVICE_INFO_KEY:@"{\"isCallerAppManaged\":\"1\",\"mdm_id\":\"mdmId\",\"object_id\":\"objectId\"}"
```

Comment:
Can you also add a test where this value is missing (like when it's coming from old broker) to test backward compatibility?

Reply:
Added